probe-rs/src/architecture/arm/core/armv7a.rs

```diff
@@ -853,6 +853,18 @@ impl<'probe> MemoryInterface for Armv7a<'probe> {
         self.execute_instruction_with_result(instr)
     }
 
+    fn read_word_16(&mut self, address: u64) -> Result<u16, Error> {
+        // Find the word this is in and its byte offset
+        let byte_offset = address % 4;
+        let word_start = address - byte_offset;
+
+        // Read the word
+        let data = self.read_word_32(word_start)?;
```

Comment:
Does this platform not support native 16bit access or is this just a shortcut? I am pretty sure we guarantee the register access sizes on those reads. I am not sure from our trait descriptions 🤔

Reply:
yeah this seemed suspicious to me too. I don't have experience with ARM-A, I just copied what `read_word_8` does.

Reply:
I agree, the accesses with explicit size should not be mapped to other access widths. The main purpose of these should be to support cases where the access width is important, and in that case it should fail if the access is not possible. Ideally with a way to query what access widths are possible.

Reply:
What put me off a bit and made me unsure is that for 32bit and 64bit we have special real HW read/writes. And I recall that we once made ones that are "optimal" and ones that are mapping HW access. So I am not sure where we drew the line. When we figure out where we wanna go we should drop proper doc comments.

Reply:
the code was already like this for 8bit accesses, IMO we should address it in a separate PR, not this one.